curvefs/src/client/rpcclient/metaserver_client.cpp

```diff
@@ -28,52 +27,395 @@ namespace curvefs {
 namespace client {
 namespace rpcclient {
 
-MetaStatusCode MetaServerClientImpl::Init(const MetaServerOption &metaopt,
-                                          MetaServerBaseClient *baseclient) {
+MetaStatusCode MetaServerClientImpl::Init(
+    const ExcutorOpt &excutorOpt, std::shared_ptr<MetaCache> metaCache,
+    std::shared_ptr<ChannelManager<MetaserverID>> channelManager) {
+    opt_ = excutorOpt;
+    metaCache_ = metaCache;
+    channelManager_ = channelManager;
     return MetaStatusCode::OK;
 }
 
+#define RPCTask                                                                \
+    [&](uint32_t poolID, uint32_t copysetID, uint32_t partitionID,             \
+        uint64_t applyIndex, brpc::Channel * channel,                          \
+        brpc::Controller * cntl) -> int
 
 MetaStatusCode MetaServerClientImpl::GetDentry(uint32_t fsId, uint64_t inodeid,
                                                const std::string &name,
                                                Dentry *out) {
-    return MetaStatusCode::OK;
+    auto task = RPCTask {
+        GetDentryResponse response;
+        GetDentryRequest request;
+        request.set_poolid(poolID);
+        request.set_copysetid(copysetID);
+        request.set_partitionid(partitionID);
+        request.set_fsid(fsId);
+        request.set_parentinodeid(inodeid);
+        request.set_name(name);
+        // TODO(lixiaocui): needset txId
+        request.set_txid(1);
+
+        curvefs::metaserver::MetaServerService_Stub stub(channel);
+        stub.GetDentry(cntl, &request, &response, nullptr);
+
+        if (cntl->Failed()) {
+            LOG(WARNING) << "GetDentry Failed, errorcode = "
+                         << cntl->ErrorCode()
+                         << ", error content:" << cntl->ErrorText()
+                         << ", log id = " << cntl->log_id();
+            return -cntl->ErrorCode();
+        }
+
+        MetaStatusCode ret = response.statuscode();
+        if (ret != MetaStatusCode::OK) {
+            LOG(WARNING) << "GetDentry: fsId = " << fsId
+                         << ", inodeid = " << inodeid << ", name = " << name
+                         << ", errcode = " << ret
+                         << ", errmsg = " << MetaStatusCode_Name(ret);
+        } else if (response.has_dentry()) {
+            *out = response.dentry();
```

Comment:
if ret = MetaStatusCode::OK,  response must has dentry,  if not have dentry at least return a error code.

Reply:
fix